### PR TITLE
session_test, executor_test: move tests of schema change.

### PR DESCRIPTION
**Pull request by hanfei1991**
move TestSchemaCheckerSQL from executor_test to session_test. @zimulala 
remove TestCommitWhenSchemaChanged which can be replaced.
move TestPrepareStmtCommitWhenSchemaChanged @tiancaiamao

### CODE REVIEW

File: new_session_test.go [hunk 1]
```diff
@@ -222,3 +235,85 @@ func (s *testSessionSuite) TestAutocommit(c *C) {
...
+	tk.MustExec(`create table t (id int, c int);`)
+	tk.MustExec(`create table t1 (id int, c int);`)
+	// insert data
+	tk.MustExec(`insert into t values(1, 1);`)
+
+	// The schema version is out of date in the first transaction, but the SQL can be retried.
+	tk.MustExec(`begin;`)
+	tk1.MustExec(`alter table t add index idx(c);`)
+	time.Sleep(lease + 2*time.Millisecond)
+	tk.MustExec(`insert into t values(2, 2);`)
```
Comment (hanfei1991) [inline new_session_test.go]: the old test is t1. It's better to use t, which can prove we have failed but retried without raising any error.

File: session_test.go [hunk 2]
```diff
@@ -2394,47 +2394,3 @@ func (s *testSessionSuite) TestRetryResetStmtCtx(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(se.AffectedRows(), Equals, uint64(1))
 }
-
-func (s *testSessionSuite) TestCommitWhenSchemaChanged(c *C) {
```
Comment (tiancaiamao) [inline session_test.go]: why remove this?

Comment (hanfei1991) [reply session_test.go]: This test is skipped and it's too simple.

Comment (tiancaiamao) [reply session_test.go]: We've meet bug related to this, to avoid repeat the same mistake again, it should not be removed.

**Comment by hanfei1991** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM @zimulala

**Comment by zimulala** [comment]
LGTM

**Pull request merged**